src/blocker.rs

```diff
@@ -234,12 +243,31 @@ impl Blocker {
             }
         });
 
+        let redirect_url: Option<String> = redirect_url_filter.as_ref().and_then(|f| {
+            // Filter redirect option is set
+            // Filter redirect option is set
+            if let Some(redirect) = f.redirect_url.as_ref() {
+                // Should be a valid URL
+                let maybe_parsed_url = parse_url(&redirect);
+                if maybe_parsed_url.is_some() {
+                    Some(maybe_parsed_url.unwrap().url)
+                } else {
+                    // Invalid URL!
+                    eprintln!("Matched rule with redirect-url option but URL not formatted correctly");
```

Comment:
@antonok-edm do you think this should error out if the passed in value for `redirect-url=` is not parsed as a URL? We don't seem to do that elsewhere in this function (check TODO on line 236)

Reply:
could you move this check to occur at list parsing time? The more we can avoid parsing things during matching the better.

Reply:
Actually nvm, I'm going to move this parse logic to network so it errors out on filter parse time, not blocker match time.

Reply:
Updated!